llvm/lib/Transforms/Scalar/TailRecursionElimination.cpp

```diff
@@ -509,7 +509,7 @@ void TailRecursionEliminator::createTailRecurseLoopHeader(CallInst *CI) {
   BasicBlock *NewEntry = BasicBlock::Create(F.getContext(), "", &F, HeaderBB);
   NewEntry->takeName(HeaderBB);
   HeaderBB->setName("tailrecurse");
-  BranchInst *BI = BranchInst::Create(HeaderBB, NewEntry);
+  BranchInst::Create(HeaderBB, NewEntry);
   // If the new branch preserves the debug location of CI, it could result in
   // misleading stepping, if CI is located in a conditional branch.
   // So, here we don't give any debug location to BI.
```

Comment:
Since there is no "BI" variable anymore it's confusing to reference it in the comment.
Perhaps change "BI" to "the new branch"?

Reply:
Refinement is committed.